mesonbuild/backend/vs2010backend.py

```diff
@@ -118,67 +118,73 @@ def __init__(self, build: T.Optional[build.Build], interpreter: T.Optional[Inter
     def get_target_private_dir(self, target):
         return os.path.join(self.get_target_dir(target), target.get_id())
 
+    def generate_genlist_for_target(self, genlist, target: build.BuildTarget, parent_node, generator_output_files, custom_target_include_dirs, custom_target_output_files) -> None:
+        if isinstance(genlist, build.GeneratedList):
+            for x in genlist.depends:
+                self.generate_genlist_for_target(x, target, parent_node, [], [], [])
+        target_private_dir = self.relpath(self.get_target_private_dir(target), self.get_target_dir(target))
+        down = self.target_to_build_root(target)
+        if isinstance(genlist, (build.CustomTarget, build.CustomTargetIndex)):
+            for i in genlist.get_outputs():
+                # Path to the generated source from the current vcxproj dir via the build root
+                ipath = os.path.join(down, self.get_target_dir(genlist), i)
+                custom_target_output_files.append(ipath)
+            idir = self.relpath(self.get_target_dir(genlist), self.get_target_dir(target))
+            if idir not in custom_target_include_dirs:
+                custom_target_include_dirs.append(idir)
+        else:
+            generator = genlist.get_generator()
+            exe = generator.get_exe()
+            infilelist = genlist.get_inputs()
+            outfilelist = genlist.get_outputs()
+            source_dir = os.path.join(down, self.build_to_src, genlist.subdir)
+            idgroup = ET.SubElement(parent_node, 'ItemGroup')
+            for i, curfile in enumerate(infilelist):
+                if len(infilelist) == len(outfilelist):
```

Comment:
> Is it important that `infilelist` and `outfilelist` are the same length, or is it just to prevent errors? Otherwise, you could extend `outfilelist` to be long enough, and do something like:

I did not wrote that part, I just moved it into a different function.
I fixed everything else you said.